Automattic/wp-calypso
# Delete unused dependencies

#### Changes proposed in this Pull Request

Delete unused packages from `package.json`. This is a first pass, many of the dependencies left can be removed once we move it down to the package that actually uses them.

I used this bash script to get a list of the candidates:

```
# Get all dependencies
for dependency in $(jq ".dependencies|keys[]" < package.json);do
	# Remove quotes 
	DEP=$(echo $dependency | sed 's/"//g' );

	# Search for the dependency in other package.json files. This are packages that already declare
	# this dependency. Then compose a partial regexp like:
	# 	package/lib1|package/lib2|app/app1
	PACKAGES=$(for file in $(git grep -l "$DEP" */**/package.json); do dirname $file; done | xargs | sed 's/ /|/g'); 

	echo $DEP;

	# Find usages of the dependency excluding:
	# 	- yarn.lock and package.json
	#	- .yarn
	#	- Files within a package that already declare this dependency
	git grep -l "$DEP" | grep -v -E "($PACKAGES)" | grep -v yarn.lock | grep -v package.json | grep -v '.yarn' ;
	echo;
done
```

It will list each dependency and files that reference it that are not inside a sub-package that already declares the dependency. For example, if both `package.json` and `client/package.json` have the dependency `myLib`, and the usages of `myLib` are all in files inside `client/...`, then `myLib` is a candidate to be removed from `package.json`.

Once I had the list of dependencies that are not used anywhere else, I excluded some of them that are used under a different name (eg: eslint plugins, babel plugins), @types or binaries used in `package.json`

#### Testing instructions

* Check that all tests passes.
* Verify there are no changes reported by ICFY
* Checkout the branch and try your usual workflow (yarn build, yarn start...)
